### PR TITLE
Add sig-ibmcloud Charter

### PR DESCRIPTION
**Pull request by spzala**
Add charter for IBMCloud SIG. This PR is reviewed and discussed in the SIG meeting.

### CODE REVIEW

**Comment by spzala** [comment]
/hold
Need to hold because specific reviewers are not assigned for the SIG yet.
/cc @spiffxp @duglin @bradtopol

File: sig-ibmcloud/charter.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [sig-governance].
+
+## Scope
+
+The IBMCloud SIG enables activities and discussion around building, deploying, maintaining, supporting,
+and using Kubernetes, and related CNCF projects, on IBM Public and Private Clouds.
+
+The authoritative source for SIG information is the [SIG README] file.
```
Comment (duglin) [inline sig-ibmcloud/charter.md]: shouldn't SIG README point to something?

File: sig-ibmcloud/charter.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+### Out of scope
+
+* Discussion around commercial aspects of IBM Public and Private clouds.
+
+## Roles and Organization Management
+
+This SIG adheres to the Roles and Organization Management outlined in [sig-governance]
```
Comment (duglin) [inline sig-ibmcloud/charter.md]: make sig-governance a link

File: sig-ibmcloud/charter.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+and opts-in to updates and modifications to [sig-governance], with exception that the SIG only
```
Comment (duglin) [inline sig-ibmcloud/charter.md]: here too

File: sig-ibmcloud/charter.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# SIG IBMCloud Charter
+
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [sig-governance].
```
Comment (duglin) [inline sig-ibmcloud/charter.md]: make sig-governance a link

File: sig-ibmcloud/charter.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+### Out of scope
+
+* Discussion around commercial aspects of IBM Public and Private clouds.
+
+## Roles and Organization Management
+
+This SIG adheres to the Roles and Organization Management outlined in [sig-governance]
+and opts-in to updates and modifications to [sig-governance], with exception that the SIG only
+have Chair role at present.
```
Comment (duglin) [inline sig-ibmcloud/charter.md]: s/have/has/

File: sig-ibmcloud/charter.md (hunk 1 above)
Comment (spiffxp) [inline sig-ibmcloud/charter.md]: Nope, please list it here.  Your README is barren as far as scope.  Specifically, please list:
- Code, Binaries, Services
- Cross-cutting and Externally Facing Processes

File: sig-ibmcloud/charter.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+* Discussion around commercial aspects of IBM Public and Private clouds.
+
+## Roles and Organization Management
+
+This SIG adheres to the Roles and Organization Management outlined in [sig-governance]
+and opts-in to updates and modifications to [sig-governance], with exception that the SIG only
+has the Chair role at present.
+
+- SIG meets every other week on Zoom at 2 PM ET on Wednesday.
```
Comment (spiffxp) [inline sig-ibmcloud/charter.md]: Out of scope for the charter doc

File: sig-ibmcloud/charter.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+### Out of scope
+
+* Discussion around commercial aspects of IBM Public and Private clouds.
+
+## Roles and Organization Management
+
+This SIG adheres to the Roles and Organization Management outlined in [sig-governance]
+and opts-in to updates and modifications to [sig-governance], with exception that the SIG only
+has the Chair role at present.
```
Comment (spiffxp) [inline sig-ibmcloud/charter.md]: Does this mean there are no duties for a Tech Lead to perform? Or that the Chair satisfies the duties of Tech Lead?

File: sig-ibmcloud/charter.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+- SIG meets every other week on Zoom at 2 PM ET on Wednesday.
+    - Agenda and meeting link
+      [here](https://goo.gl/f9EbvY).
+    - Recordings of the calls are made available
+      [here](https://goo.gl/PSkwBj).
+    - SIG mailing list link
+      [here](https://goo.gl/hUF9eF).
+
```
Comment (spiffxp) [inline sig-ibmcloud/charter.md]: You're missing the section on Subproject Creation.

File: sig-ibmcloud/charter.md (hunk 7 above)
Comment (spzala) [reply sig-ibmcloud/charter.md]: Yes, the chair role satisfies the duties as of today. We don't have code repo etc yet but once we have we will rethink on adding Tech lead role. Hope makes sense.

File: sig-ibmcloud/charter.md (hunk 6 above)
Comment (spzala) [reply sig-ibmcloud/charter.md]: I am fine removing it but I used some of the already merged charter as reference and charters like sig-azure, cloud-provider, service-catalog (https://github.com/kubernetes/community/blob/master/sig-service-catalog/charter.md) have a similar addition. Also, as most SIGs charter (and charter template) has reference to the sig-governance (https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md) where it has a similar mention on meeting schedule, so my understanding was in the charter we provide a concise meeting detail etc. Again, I am fine to remove it if we decide so. Thanks!

File: sig-ibmcloud/charter.md (hunk 8 above)
Comment (spzala) [reply sig-ibmcloud/charter.md]: We don't have any subproject yet so it's not mentioned. I am fine if it's good to add a section with `none` for now? Thanks!

File: sig-ibmcloud/charter.md (hunk 1 above)
Comment (spzala) [reply sig-ibmcloud/charter.md]: We don't have code repos etc yet as I mentioned in another comment as well, but creating provider repo is a work in progress right now and we will be updating charter once it happens. I think we can safely remove the line mentioning README, and will do it. Thanks!

File: sig-ibmcloud/charter.md (hunk 8 above)
Comment (spiffxp) [reply sig-ibmcloud/charter.md]: This is about documenting the process you will follow when you make a subproject.  You need to choose [an option from the template](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#subproject-creation):
- Option 1: by SIG Technical Leads
- Option 2: by Federation of Subprojects

Also, I'm not entirely sure how on board I am with the notion of a SIG that owns no artifacts within this project.  I guess I can find no prior art for this expectation other than ["areas the SIG is responsible for directing and maintaining"](https://github.com/kubernetes/community/blob/master/committee-steering/governance/README.md)

Is there code you own?  Are there docs?

File: sig-ibmcloud/charter.md (hunk 7 above)
Comment (spiffxp) [reply sig-ibmcloud/charter.md]: Can add a sentence stating this explicitly? eg: "Chairs must also fulfill all of the responsibilities of the Tech Lead role as outlined in [sig-governance]" (with a link)

File: sig-ibmcloud/charter.md (hunk 6 above)
Comment (spiffxp) [reply sig-ibmcloud/charter.md]: You managed to pick as prior art three charters that merged either without much SC review, or before we got stricter about sticking to a common template.  We plan on ensuring those change to match the template.

You can check https://github.com/kubernetes/steering/issues/31 to see more info

_That_ you meet regularly is an expectation that doesn't need to be spelled out here.  When and how should be spelled out in sigs.yaml

File: sig-ibmcloud/charter.md (hunk 1 above)
Comment (spiffxp) [reply sig-ibmcloud/charter.md]: I would look to [SIG VMWare's in-flight charter](*https://github.com/kubernetes/community/pull/2300) for some prior art here, with the caveat that I currently have some unaddressed nits there.  I think they do a good job of taking ownership of community management and best practices in a way that articulates some artifacts and processes.

File: sig-ibmcloud/charter.md (hunk 8 above)
Comment (spzala) [reply sig-ibmcloud/charter.md]: I see, I thought it's something related to existing sub projects. We don't have one yet but we have discussed this in SIG. We have provider code but not yet public and it's WIP to have a pubic repo. Thanks!

**Comment by spiffxp** [comment]
Ping, any updates on this?

**Comment by spzala** [comment]
@spiffxp hello, sorry I was OOO, just created second commit. Thanks!

File: sig-ibmcloud/charter.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+- Determining and documenting best practices for configuring Kubernetes on IBM Cloud Private (ICP).
+- Discussing IKS and ICP tracking of Kubernetes features and releases.
+- Utilizing Kubernetes and related CNCF projects (e.g. Helm, Istio) by IKS and ICP.
+- Discussing bugs and feature requests recorded as Kubernetes upstream issues on GitHub. These issues should be tagged with `sig/ibmcloud`.
+
+#### Code, Binaries and Services
+
+The work to have a cloud provider specific public code repository is in progress. This section will be updated once the work is complete. Kubernetes upstream code that
+is directly related to IKS or ICP issues or features can be discussed.
+
```
Comment (spiffxp) [inline sig-ibmcloud/charter.md]: You're missing the "Cross-cutting and Externally Facing Processes" section. I assume this means you have none.

**Comment by spiffxp** [comment]
/lgtm
That said, this looks really light.  There are currently 5 other SIGs that have no subprojects listed, so you're not alone on this, but I anticipate we'll want to resolve this down the line.

/assign @derekwaynecarr 
you're the other steering committee member for this

File: sig-ibmcloud/charter.md (hunk 9 above)
Comment (spzala) [reply sig-ibmcloud/charter.md]: Yes, you got it right :) Thanks!

**Comment by spzala** [comment]
@spiffxp, thanks Aaron!! Yes, agree, we should have one or more in future as we work on open sourcing the provider code.

File: sig-ibmcloud/charter.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+* Internal or commercial aspects of IKS and ICP.
+
+## Roles and Organization Management
+
+This SIG adheres to the Roles and Organization Management outlined in [sig-governance]
+and opts-in to updates and modifications to [sig-governance], with exception that the SIG only
+has the Chair role at present. Chairs must also fulfill all of the responsibilities of the Tech Lead role as outlined in [sig-governance].
+
+### Subproject Creation
+Associated subprojects are created following the `by SIG Technical Leads` option procedure described in [sig-governance].
```
Comment (derekwaynecarr) [inline sig-ibmcloud/charter.md]: nit: you can just say "SIG Technical Leads" in this section.

as an example: https://github.com/kubernetes/community/pull/2462/files#diff-ddf39acdb13e6743dcb3d690a0a68f96R60

Comment (derekwaynecarr) [reply sig-ibmcloud/charter.md]: i assume creation of subprojects will also be impacted by changes in "code, binaries, services" discussion.

**Comment by derekwaynecarr** [comment]
I do not have a great sense on the scope of subprojects that could potentially be sponsored by this sig since there really are none yet.  i would like to revisit this in the future once the code, binaries, and services section is better fleshed out.  i agree w/ aaron that we can resolve this down the line.

/hold cancel
/lgtm
/approve

**Pull request merged**